src/coreclr/tools/aot/ILCompiler.Compiler/IL/ILImporter.Scanner.cs

```diff
@@ -974,13 +974,36 @@ private void ImportReadOnlyPrefix()
             _isReadOnly = true;
         }
 
-        private void ImportFieldAccess(int token, bool isStatic, string reason)
+        private void ImportFieldAccess(int token, bool isStatic, bool? write, string reason)
```

Comment:
Does this analysis interact correctly with unsafe code?

E. g. `&s + <offset> = <value>;`

Reply:
It doesn't, and neither with interop.

Reply:
This should not be a problem in practice, given what we use the result of the analysis for today. Correct?

Reply:
After 865875849d186d08a071f62aaa84c51cf30f966a I can't think of a problematic case - this should not be making any new assumptions.

Reply:
Are the consumers restricted enough that even if the field is only written to via pointer arithmetic, things still work?

```cs
[StructLayout(LayoutKind.Explicit)]
class Xyz
{
    [FieldOffset(0)]
    public int FieldOne;
    [FieldOffset(4)]
    public int FieldTwo;
}

Xyz xyz = { ... };
Unsafe.AddByteOffset(ref xyz.FieldOne, 4) = { ... };
```

Reply:
We go over all fields on the class and check for read only on each. The check is in the preexisting code that we're switching from believing isliteral to using this analysis (that's the correctness fix part from the pr description).